lib/ims/lti/outcome_response.rb

```diff
@@ -105,18 +105,22 @@ def has_error?
 
     # Parse Outcome Response data from XML
     def process_xml(xml)
-      doc = REXML::Document.new xml
-      @message_identifier = doc.text("//imsx_statusInfo/imsx_messageIdentifier").to_s
-      @code_major = doc.text("//imsx_statusInfo/imsx_codeMajor")
-      @code_major.downcase! if @code_major
-      @severity = doc.text("//imsx_statusInfo/imsx_severity")
-      @severity.downcase! if @severity
-      @description = doc.text("//imsx_statusInfo/imsx_description")
-      @description = @description.to_s if @description
-      @message_ref_identifier = doc.text("//imsx_statusInfo/imsx_messageRefIdentifier")
-      @operation = doc.text("//imsx_statusInfo/imsx_operationRefIdentifier")
-      @score = doc.text("//readResultResponse//resultScore/textString")
-      @score = @score.to_s if @score
+      begin
+        doc = REXML::Document.new xml
+        @message_identifier = doc.text("//imsx_statusInfo/imsx_messageIdentifier").to_s
+        @code_major = doc.text("//imsx_statusInfo/imsx_codeMajor")
+        @code_major.downcase! if @code_major
+        @severity = doc.text("//imsx_statusInfo/imsx_severity")
+        @severity.downcase! if @severity
+        @description = doc.text("//imsx_statusInfo/imsx_description")
+        @description = @description.to_s if @description
+        @message_ref_identifier = doc.text("//imsx_statusInfo/imsx_messageRefIdentifier")
+        @operation = doc.text("//imsx_statusInfo/imsx_operationRefIdentifier")
+        @score = doc.text("//readResultResponse//resultScore/textString")
+        @score = @score.to_s if @score
+      rescue => e
```

Comment:
Can you scope the error to be specific to the XML parse error you are expecting?